src/Settings/View/AbstractView.php

```diff
@@ -0,0 +1,86 @@
+<?php
+
+declare(strict_types=1);
+
+namespace MyParcelNL\Pdk\Settings\View;
+
+use InvalidArgumentException;
+use MyParcelNL\Pdk\Base\Support\Arrayable;
+use MyParcelNL\Pdk\Base\Support\Collection;
+use MyParcelNL\Pdk\Base\Support\Utils;
+
+abstract class AbstractView implements Arrayable
+{
+    /**
+     * @var array
+     */
+    protected $data;
+
+    /**
+     * @var array
+     */
+    protected $fields;
+
+    /**
+     * @param  array $data
+     */
+    public function __construct(array $data)
+    {
+        $this->data = $data;
+    }
+
+    /**
+     * @return \MyParcelNL\Pdk\Base\Support\Collection
+     */
+    abstract protected function getFields(): Collection;
+
+    /**
+     * @return array
+     */
+    public function toArray(): array
+    {
+        if (! isset($this->fields)) {
+            $this->fields = $this->getFields();
+
+            if (! empty($this->data)) {
+                $this->fields->push($this->data);
+            }
+        }
+
+        return $this
```

Comment:
het punt was dat dit niet meerdere keren uitgevoerd hoeft te worden omdat fields niet meer verandert. dit is nu niet gecached